### ISSUE TITLE
Cache for getBlock

### ISSUE THREAD

**Issue by AlfredoG87** [hashgraph/hedera-json-rpc-relay#1422]
### Problem

Currently we have a cache at the method level for `eth_getBlockByHash` and `eth_getBlockByNumber`
They both use the method `getBlock`.

So the issue I see is that imagine distinct users get the same block, just that some get it byHash and others byNumber, we would have 2 copies of the same cached block and also we would need to compute the same info, potencially doing hundreds of calls to the mirror node to get the details when marked that details are needed.

### Solution

Remove the cache at the method level and move it at the level of `getBlock` 

have 1 cache, 1 for the block by hash or block by number without the details.

have another cache for the block details that is only by block number, once we get the general block by hash, we know the number, so we are able to cache or get the cache using the number, even if the request was by hash.

build the block and return it.

**Comment by Nana-EC** [comment]
I think this was resolved as both eventually call `mirrorNoceClient.getBlocks` which has a cache.
@AlfredoG87 can you confirm and close if so

**Comment by AlfredoG87** [comment]
@Nana-EC no, that is not the case (see [here](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/clients/mirrorNodeClient.ts#L575-L591)), we are not caching at the level of the mirror node, however, we might want to reconsider if we really need to set a cache at the mirror node client level as originally proposed, considering that the mirror-node itself does has a cache, so even if we repeat the request, the only cost will be the network latency, and is really low.

I would say is no longer necessary, let me know if you agree with me so we can close the ticket.

**Comment by Ferparishuertas** [comment]
Wont fix. If any action will be reviewed on a new global cache review
